### PR TITLE
refactor: fe updates due to api changes

### PR DESCRIPTION
**Pull request by omarCreativeDev**
TISNEW-4088

- refactor fe code in order to align with be endpoint schema
- added new local environment file alongside new `npm start:local` script
- updated proxy config
- update unit and e2e tests
- upgrade prettier & cypress
- cleanup npm scripts

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -103,6 +101,20 @@
               ],
               "serviceWorker": true,
               "ngswConfigPath": "ngsw-config.json"
+            },
+            "local": {
```
Comment (ojemuyiwa) [inline angular.json]: dev is environment.ts, staging environment.stage.ts, and prod. which environment is local

Comment (omarCreativeDev) [reply angular.json]: yea thats correct. local is for when running the be locally on your own machine

Comment (ojemuyiwa) [reply angular.json]: yeah thats dev

Comment (ojemuyiwa) [reply angular.json]: with debug zones enabled

Comment (omarCreativeDev) [reply angular.json]: dev `environment.ts` is for consuming FE controlled mock api's with `mocky.io` during local development.

I added a new local environment file `environment.local.ts` so provide the ability to consume locally running docker based BE during local development

- so as before to spin up app for development with mocky `npm run start`
- and to spin up app for development with local BE `npm run start:local`

**Pull request merged**